modules/tydux/src/lib/Facade.ts

```diff
@@ -204,6 +205,9 @@ export abstract class Facade<C extends Commands<S>, S = CommandsState<C>> {
     }
 
     private setState(state: S) {
+        if (isNotNil(state) && ![Array, Map, Set, Object].includes((state as any).constructor)) {
+            (state as any).constructor[immerable] = true;
+        }
```

Comment:
@romanroe I add the immerable symbol to class based states when its been set through the facade. Since we do not specify S more specific I have to do the cast. Do you know a smarter way to determine if `state: S` is a value based on a class? 

See [Object.prototype.constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/constructor) and [immerable semantics](https://immerjs.github.io/immer/complex-objects#semantics-in-detail)